### PR TITLE
Use direct wallets as params, remove ffi

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Start singularity with direct wallet key from env, rather than having a whole wallet handler in singularity

# Implementation

- removes FFI
- removes wallet code
- removes code to build ffi from makefile & dockerfile
- adds a wallet private key parameter
- uses that parameter directly with singularity

### CODE REVIEW

**Review by willscott — APPROVED** [review]
not having the dep in motion as well as singularity simplifies this layer a bunch!

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,9 @@
 SHELL=/usr/bin/env bash
 .DEFAULT_GOAL := build
```
Comment (masih) [inline Makefile:2]: Remove this file entirely?

File: integration/singularity/options.go [hunk 2]
```diff
@@ -48,10 +48,8 @@ func newOptions(o ...Option) (*options, error) {
 			return nil, err
 		}
 	}
-	if opts.wallet == nil {
-		var err error
-		opts.wallet, err = wallet.New()
-		return nil, err
+	if opts.walletKey == "" {
+		return nil, errors.New("must specify a wallet address")
```
Comment (masih) [inline integration/singularity/options.go:52]:
```suggestion
		return nil, errors.New("must specify a wallet key")
```

**Review by masih — APPROVED** [review]
We may want to revisit this for multi wallet support.

**Comment by masih** [comment]
Would be nice to avoid FFI. 

Another thing that occurred to me is that changes here also remove the ability to generate a wallet if it's not specified. 

That's fine by me as long as product folks are happy and alternative ways of generating wallet is communicated to the end users.

**Comment by hannahhoward** [comment]
@masih one thing to understand is that if you don't put money in a wallet, it doesn't exist on chain, and you can't make deals with it, even free ones. So it has to be funded anyway.

**Pull request merged**